packages/http/src/__tests__/http-prism-instance.spec.ts

```diff
@@ -172,6 +174,55 @@ describe('Http Client .process', () => {
           expect(result.output!.body).toEqual(reply);
         });
       });
+
+      describe('Prism user-agent header', () => {
+        it('adds user-agent header in the form of Prism/<<PRISM_VERSION>>', async () => {
+          const oasBaseUrl = 'http://example.com/api';
+
+          nock(oasBaseUrl)
+            .get('/pet')
+            .reply(function() {
+              expect(this.req.headers['user-agent']).toBe(`Prism/${prismVersion}`);
+
+              return [200, 'reply', {}];
+            });
+
+          await prism.process(
+            {
+              method: 'get',
+              url: {
+                path: '/pet',
+              },
+            },
+            config,
+          );
+        });
+
+        it('specifying user-agent in the options parameter has no impact on user-agent header to be sent', async () => {
```

Comment:
This user-agent should just be a default, folks can override if they want. That's how most HTTP clients handle it anyhow.